### PR TITLE
test: add test for private registry module with dir suffix

### PR DESCRIPTION
**Pull request by stigok**
I am suspecting that module sources containing a dir suffix is not properly handled. Unable to test locally due to the size of the Docker developer image.

### CODE REVIEW

**Comment by stigok** [comment]
Thanks! I've fixed my lint errors now. Could you please take a look at my test to see if it I am testing what I think I'm testing? I am trying to see if dependabot actually ends up with the actual module and registry URL even when the `source = "<src>"` contains a `//sub/dir/in/module` suffix.

```terraform
module "foo" {
    source = "registry.example.com/my/module/generic//sub-dir"
    version = "1.2.3"
}
```

**Comment by stigok** [comment]
This actually seems to work out fine. We may have some other problems in our setup outside of Dependabot. Feel free to merge this if you care about this extra test.

File: terraform/spec/fixtures/projects/private_module_with_dir_suffix/main.tf
Comment (jeffwidman) [inline terraform/spec/fixtures/projects/private_module_with_dir_suffix/main.tf:2]: I don't know `terraform` very well... is the double slash `//` in this path expected/desired?

**Review by jeffwidman — COMMENTED** [review]
Always happy to add more tests, had a quick question though before merging...

File: terraform/spec/fixtures/projects/private_module_with_dir_suffix/main.tf
Comment (stigok) [reply terraform/spec/fixtures/projects/private_module_with_dir_suffix/main.tf:2]: Yeah, this is to select a subfolder within the module source. https://developer.hashicorp.com/terraform/language/modules/sources#modules-in-package-sub-directories

**Review by jeffwidman — APPROVED** [review]

**Pull request merged**
